src/image/ImageManagerActions.cc

```diff
@@ -562,6 +595,23 @@ int ImageManager::delete_image(int iid, string& error_str)
         {
             NebulaLog::log("ImM",Log::ERROR,
                 "Image could not be removed from DB");
+
+            return -1;
+        }
+
+        int uid = img->get_uid();
+        int gid = img->get_gid();
+        long long size = img->get_size() + img->get_snapshots().total_size();
+
+        img.reset();
+
+        Quotas::ds_del(uid, gid, ds_id, size);
+
+        // Remove Image reference from Datastore
+        if ( auto ds = dspool->get(ds_id) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This snippet may contain an accidental use of incorrect operation. Make sure that the operation being performed is in fact what you intended.